clash-lib/src/Clash/Core/Type.hs

```diff
@@ -513,3 +515,33 @@ isIntegerTy :: Type -> Bool
 isIntegerTy (ConstTy (TyCon (nm)))
   | "GHC.Integer.Type.Integer" `isPrefixOf` (name2String nm) = True
 isIntegerTy _ = False
+
+-- Normalize a type, looking through Signals and newtypes
+--
+-- For example: Vec (3+2) (Unsigned (3+1)) normalizes to Vec 5 (Unsigned 4)
+normalizeType :: HashMap TyConOccName TyCon -> Type -> Type
+normalizeType tcMap = go
+  where
+  go ty = case tyView ty of
+    TyConApp tcNm args
+      | name2String tcNm == "Clash.Signal.Internal.Signal"
+      , [_,elTy] <- args
+      -> go elTy
+      | name2String tcNm == "Clash.Sized.Internal.BitVector.BitVector" ||
```

Comment:
Add a comment that you are looking through newtypes, but that clash' primitive number types are implemented as newtypes as well and you don't want to normalize them away.